eng/Version-Increment.ps1

```diff
@@ -0,0 +1,90 @@
+<#
+.SYNOPSIS
+Bumps up package versions after release
+
+.DESCRIPTION
+This script bumps up package versions following conventions defined at https://github.com/Azure/azure-sdk/blob/master/docs/policies/releases.md#incrementing-after-release-net
+
+.PARAMETER RepoRoot
+The Root of the repo
+
+.PARAMETER ServiceDirectory
+The Name of the Service Directory
+
+.PARAMETER PackageName
+The Name of the Package
+
+.EXAMPLE
+Updating package version for core
+
+Version-Increment.ps1 -PackageCsprojPath "C:\Git\azure-sdk-for-net\sdk\core\Azure.Core\src\Azure.Core.csproj"
+
+#>
+
+[CmdletBinding()]
+Param (
+    [Parameter(Position=0)]
+    [ValidateNotNullOrEmpty()]
+    [string] $RepoRoot,
+    [string] $ServiceDirectory,
+    [string] $PackageName
+)
+# Regular expression as specified in https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string
+$VERSION_REGEX = "^(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$"
+
+$CsprojData = New-Object -TypeName XML
+$PackageCsprojPath = Join-Path $RepoRoot "sdk" $ServiceDirectory $PackageName "src" "${PackageName}.csproj"
+$CsprojData.Load($PackageCsprojPath)
+
+$PackageVersion = Select-XML -Xml $CsprojData -XPath '/Project/PropertyGroup/Version'
+if ([System.String]::IsNullOrEmpty($PackageVersion))
+{
+    Write-Error "Package Version is missing"
+    exit 1
+}
+
+Write-Verbose "Current Version: ${PackageVersion}"
+
+$Major = $Minor = $Patch = $Pre = $PreTag = $PreVer = $Null
+
+if ($PackageVersion -Match $VERSION_REGEX)
+{
+    if($Matches[4] -eq $null){$Pre = @()}
+    else{$Pre = $Matches[4].Split(".")}
+
+    $Major = [int]$Matches[1]
+    $Minor = [int]$Matches[2]
+    $Patch = [int]$Matches[3]
+    $PreTag = [string]$Pre[0]
+    $PreVer = [int]$Pre[1]
+}
+else
+{
+    Write-Error "Package Version is Incorrect"
```

Comment:
Version property contains incorrect format. It should be in format "X.Y.Z[-preview.N]"